crates/ffmpeg/src/io/output.rs

```diff
@@ -281,3 +281,206 @@ impl Output<()> {
+            }
+            _ => panic!("Expected FfmpegError::Arguments"),
+        }
+    }
+
+    #[test]
+    fn test_output_into_inner() {
+        let data = Cursor::new(Vec::with_capacity(1024));
+        let options = OutputOptions::default().format_name("mp4");
+        let output = Output::new(data, options).expect("Failed to create Output");
+        let inner_data = output.into_inner();
+
+        assert!(inner_data.get_ref().is_empty());
+        let buffer = inner_data.into_inner();
+        assert_eq!(buffer.capacity(), 1024);
+    }
+
+    #[test]
+    fn test_output_new() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let output = Output::new(data, options);
+
+        assert!(output.is_ok());
+    }
+
+    #[test]
+    fn test_output_seekable() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let output = Output::seekable(data, options);
+
+        assert!(output.is_ok());
+    }
+
+    #[test]
+    fn test_output_set_metadata() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let mut output = Output::new(data, options).unwrap();
+        let metadata = Dictionary::new();
+        output.set_metadata(metadata);
+
+        assert!(!output.as_ptr().is_null());
+    }
+
+    #[test]
+    fn test_output_as_mut_ptr() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let mut output = Output::new(data, options).expect("Failed to create Output");
+        let context_ptr = output.as_mut_ptr();
+
+        assert!(!context_ptr.is_null(), "Expected non-null pointer from as_mut_ptr");
+    }
+
+    #[test]
+    fn test_add_stream_with_valid_codec() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let mut output = Output::new(data, options).expect("Failed to create Output");
+        let dummy_codec: *const AVCodec = 0x1234 as *const AVCodec;
+        let stream = output.add_stream(Some(dummy_codec));
+
+        assert!(stream.is_some(), "Expected a valid Stream to be added");
+    }
+
+    #[test]
+    fn test_copy_stream() {
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let mut output = Output::new(data, options).expect("Failed to create Output");
+        // create a new output to prevent double mut borrow
+        let data = Cursor::new(Vec::new());
+        let options = OutputOptions::default().format_name("mp4");
+        let mut output_two = Output::new(data, options).expect("Failed to create Output");
+
+        let dummy_codec: *const AVCodec = 0x1234 as *const AVCodec;
+        let source_stream = output.add_stream(Some(dummy_codec)).expect("Failed to add source stream");
```

Comment:
What does the source stream have pre-configured? I see you assert against time_base & start_time & disposition but like what values do they have when you by default create a stream. Can we modify them so that the copy actually checks if we copy the values and not just default init?